src/components/Layout/RightPaneLayout/RightPaneContext.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ReactElement, ReactNode, useContext, useMemo, useState } from "react";
+
+interface OpenRightPaneOpts {
+  content: ReactNode;
+}
+
+export type RightPaneLayoutContextProps = {
+  openInPane: (opts: OpenRightPaneOpts) => void;
+  closePane: () => void;
+  isRightPaneOpen: Boolean;
+  rightPaneContent: ReactNode;
+};
+
+export const RightPaneContext = React.createContext<RightPaneLayoutContextProps>({
+  openInPane: () => {},
+  closePane: () => {},
+  isRightPaneOpen: false,
+  rightPaneContent: null,
+});
+
+export function RightPaneProvider({ children }: { children: ReactElement }) {
+  const [rightPaneContent, setRightPaneContent] = useState<ReactNode>(undefined);
+  const [isRightPaneOpen, setIsRightPaneOpen] = useState<Boolean>(false);
+
+  const context = useMemo(() => {
+    return {
+      openInPane: (opts: OpenRightPaneOpts) => {
+        setRightPaneContent(opts?.content);
+        setIsRightPaneOpen(true);
+      },
+      closePane: () => setIsRightPaneOpen(false),
```

Comment:
**thought:** If we do want to keep both the states (`rightPaneContent` and `isRightPaneOpen`), should we setting `rightPaneContent` back to `undefined` here? 
I could see the argument that you don't want to have the content removed until after the animation to close has finished, so maybe we can figure that out too, but I'd be okay with just having it removed.

Reply:
I agree It should be cleared after the close animation has been performed.

Reply:
Note: framer-motion exposes a `onAnimationComplete` which is nice to check against the pane being open vs closed for clearing content